hilbert/runners/run_base.py

```diff
@@ -51,11 +61,17 @@ def run(solver_factory, **args):
     # Train train train!  Write to disk once in awhile!
     updates_per_write = int(num_updates / num_writes)
     for write_num in range(num_writes):
-        solver.cycle(updates_per_write, monitor_closely)
+        try:
+            solver.cycle(updates_per_write, monitor_closely)
+        except h.exceptions.DivergenceError:
+            print("\n\nDied at {} epoch.".format(write_num))
+            tracer.declare(key='wrote_num', value=write_num+1)
+            raise h.exceptions.DivergenceError("Model has diverged")
```

Comment:
use either `print` or `tracer.declare`, not both!  Declare is better, because it prints and it logs to the trace.  You can bypass the key/value format for tracer if you just want to print anything, by using `tracer.trace()`.